app.go

```diff
@@ -618,9 +672,18 @@ func (app *App) provide(p provide) {
 
 	switch {
 	case p.IsSupply:
-		app.logger.PrintSupply(constructor)
+		// app.log.PrintSupply(constructor)
+		for _, rtype := range fxreflect.ReturnTypes(constructor) {
+			fxlog.Info("supplying", fxlog.Field{Key: "constructor", Value: rtype}).Write(app.log)
+		}
 	default:
-		app.logger.PrintProvide(constructor)
+		// app.log.PrintProvide(constructor)
+		for _, rtype := range fxreflect.ReturnTypes(constructor) {
+			fxlog.Info("providing",
+				fxlog.Field{Key: "return value", Value: rtype},
```

Comment:
field names should be alphanumeric without spaces. these are typically json object keys that are ideally addressable in the form `obj.foo`. so, `returnValue`, not `return value`.